src/core/execute.c

```diff
@@ -455,8 +450,8 @@ int exec_spawn(
 
         /* The executor binary is pinned, to avoid compatibility problems during upgrades. */
         r = posix_spawn_wrapper(
-                        FORMAT_PROC_FD_PATH(unit->manager->executor_fd),
-                        STRV_MAKE(executor_path,
+                        SYSTEMD_EXECUTOR_BINARY_PATH,
```

Comment:
I think this breaks using the locally built binary in tests, no? pin_callout_binary() took care of that